app/controllers/admin/reports_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Admin::ReportsController < Admin::BaseController
+  def index
+  end
+
+  def deals
+    @deals = Deal.deals_with_revenue
+    flash[:notice] = 'No customers found' if @deals.empty?
+  end
+
+  def customers
+    if params[:from].present? && params[:to].present?
+      @customers = User.customers_orders(params[:from]&.to_datetime, params[:to].to_datetime)
+    else
+      @customers = User.customers_orders
+    end
```

Comment:
What if only one is provided. 
What if i send only `from` and want to have orders from that date till present or maybe all the orders from beginning to a specific date
Handle all cases please